### PR TITLE
CICD pipeline github actions

### PR DESCRIPTION
**Pull request by priyaranjanpatil**
-Currently CI/CD pipeline built for only DEV env
-Latest version will be auto deployed and also can rollback to previous versions
-We are using Dockerfile to build and push images to ECR & ECS also Env vars will be part of ECS task defnition
-Docker image scan is added in the pipeline which is using Trivy vulnerability scanner
-Environment variables
Platform specific: are stored in AWS SSM paramater store and pulled in Github actions while running pipeline
App specific: are stored in Github actions Env variables

### CODE REVIEW

**Review by nandeeshsb9 — APPROVED** [review]

**Review by vietnguyengit — CHANGES_REQUESTED** [review]
Please clean up the commit history

File: .dockerignore [hunk 1]
```diff
@@ -43,4 +43,4 @@ buildNumber.properties
 # JDT-specific (Eclipse Java Development Tools)
 .classpath
 
-.env
+.env
```
Comment (vietnguyengit) [inline .dockerignore:46]: missing ending line

**Review by vietnguyengit — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -19,4 +19,4 @@ COPY --from=builder /app/target/es-indexer-0.0.1-SNAPSHOT.jar /app/app.jar
 
 EXPOSE 8080
 
-CMD ["java", "-jar", "app.jar"]
+CMD ["java", "-jar", "app.jar"]
```
Comment (vietnguyengit) [inline Dockerfile:22]: missing ending line

File: .github/workflows/github-actions.yml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            ELASTICSEARCH_INDEX_NAME=${{ vars.ELASTICSEARCH_INDEX_NAME }}
+            IMAGE=${{ steps.push-image.outputs.image }}
+
+      - name: Deploy Amazon ECS task definition
+        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+        with:
+          task-definition: ${{ steps.task-def.outputs.task-definition }}
+          service: ${{ env.ECS_SERVICE }}
+          cluster: ${{ env.ECS_CLUSTER }}
+          wait-for-service-stability: true
```
Comment (vietnguyengit) [inline .github/workflows/github-actions.yml:192]: missing ending line

**Review by vietnguyengit — CHANGES_REQUESTED** [review]
also please put back the `ecr-build,yml` if this PR won't immediately have the ECS instance available @priyaranjanpatil

 removing `ecs-build.yml` will disrupt current process for refreshing the indexer instance we need.

File: aodn-es-indexer-td.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        "FARGATE"
+    ],
+    "cpu": "1024",
+    "memory": "3072",
+    "runtimePlatform": {
+        "cpuArchitecture": "X86_64",
+        "operatingSystemFamily": "LINUX"
+    },
+    "tags": []
+}
```
Comment (vietnguyengit) [inline aodn-es-indexer-td.json:75]: missing ending line

**Comment by priyaranjanpatil** [comment]
closing this PR and will open another one after resolving merge and rebase to main

**Comment by vietnguyengit** [comment]
thanks @priyaranjanpatil that's good, rebasing should be with `git rebase` command always, please also organise the commits history when you work on new branch and create new PR, you don't need to push every new commits you have locally, which will create a mess in git log unnecessarily, but if they are for github action, which you have to push to remote branch to see actual affects, when you are happy with changes you've made:

your can run commands like these to rewrite the commits history with `git reset --soft HEAD~<number of commits>`:

or you can do git squash with iterative mode to pick and squash whatever commits in the log

**Pull request closed without merging**